criu/mount-v2.c

```diff
@@ -933,12 +933,19 @@ static int move_mount_set_group(int src_id, char *source, int dst_id)
 
 static int restore_one_sharing(struct sharing_group *sg, struct mount_info *target)
 {
-	char target_path[PATH_MAX];
+	char tfd_path[PATH_MAX] = {0};
+	char target_path[PATH_MAX] = {0};
 	int target_fd;
 
 	target_fd = fdstore_get(target->mnt_fd_id);
 	BUG_ON(target_fd < 0);
-	snprintf(target_path, sizeof(target_path), "/proc/self/fd/%d", target_fd);
+	snprintf(tfd_path, sizeof(tfd_path), "/proc/self/fd/%d", target_fd);
+
+	if (readlink(tfd_path, target_path, sizeof(target_path)) < 0) {
```

Comment:
Can you explain how this helps with https://github.com/torvalds/linux/commit/12f147ddd6de7382dad54812e65f3f08d05809fc check? `check_mnt(mnt)` operates on a mount level, and procfs symlink should be resolved by this time already inside the kernel.

Reply:
> This fix doesn't look like the right one. We want to do changes on the specific mount, so we need to enter its mount namespace and do changes from it.

So, we this usecase is required in criu. Maybe kernel check patch is wrong.